rotkehlchen/chain/evm/node_inquirer.py

```diff
@@ -975,9 +975,9 @@ def _get_logs(
             to_block=to_block,
         )
 
-        if event_abi['anonymous']:
+        if event_abi.get('anonymous', False):
             # web3.py does not handle the anonymous events correctly and adds the first topic
-            filter_args['topics'] = filter_args['topics'][1:]
+            filter_args['topics'] = filter_args['topics'][1:]  # pyright: ignore  # I think FilterParams is not well defined. It always has topics
```

Comment:
This FilterParams one is defined with all of them optional.  https://github.com/ethereum/web3.py/blob/main/web3/types.py#L302

Though I have always seen it exist. So I think it's just a bad definition?